### DISCUSSION TITLE
Upload Books to OpenLibrary

### DISCUSSION THREAD

**Discussion by PrasannaVenkadesh** [mateusz-bak/openreads#700] [Ideas]
I would like to contribute to OpenLibrary via Openreads. Not all books are in OpenLibrary. If I add a book manually in Openreads, I would like to see a button to upload this book to OpenLibrary, provided I use my login credentials.

**Comment by mateusz-bak** [comment]
Hello @PrasannaVenkadesh thank you for the suggestion. I looked through the Open Library's API and haven't found anything for submitting books. I'll talk with Open Library team about this and check if we can do this.

**Comment by PrasannaVenkadesh** [comment]
I believe [this wiki](https://github.com/internetarchive/openlibrary/wiki/Endpoints) of API endpoints should help. But I could see they are also incomplete. For example, I could see endpoint to create a work, but endpoint for creating a edition is missing.

**Comment by BoboTiG** [comment]
Having this feature would be huge. We would all be able to contribute to OpenLibrary, and thus it would become a Wikipedia/Wiktionary-like project (and we would finally be able to quit Goodreads & co). Coupled to #54, and we have a fully featured "Internet Archive" project :)

**Comment by lucastucious** [comment]
We would need some additional entries before that, as suggested in #330

**Reply by eclecticpassions** [reply]
Sorry to bring up an old thread, but I can't find and Android app that can upload to OpenLibrary... Is it achievable now from their API? Thanks.

**Reply by lucastucious** [reply]
accoridng to this doc https://openlibrary.org/developers/api it is. But i cannot find any app that can do it. I hope [Openshelf](https://github.com/Swappppy/Openshelf) will do it ?

**Reply by eclecticpassions** [reply]
I tried installing Openshelf but there's no export option or sync to OpenLibrary. I'll try to ask over there, thanks. I'd really want an OpenLibrary Android app to track my reading! Thanks Luca.
